services/web/server/src/simcore_service_webserver/folders/_folders_db.py

```diff
@@ -467,6 +463,60 @@ async def get_projects_recursively_only_if_user_is_owner(
         return [ProjectID(row[0]) async for row in result]
 
 
+async def get_all_folders_and_projects_recursively(
+    app: web.Application,
+    connection: AsyncConnection | None = None,
+    *,
+    folder_id: FolderID,
+    private_workspace_user_id_or_none: UserID | None,
+    product_name: ProductName,
+) -> tuple[list[FolderID], list[ProjectID]]:
+    """
+    The purpose of this function is to retrieve all projects within the provided folder ID.
+    """
+
+    async with pass_or_acquire_connection(get_asyncpg_engine(app), connection) as conn:
+
+        # Step 1: Define the base case for the recursive CTE
+        base_query = select(
+            folders_v2.c.folder_id, folders_v2.c.parent_folder_id
+        ).where(
+            (folders_v2.c.folder_id == folder_id)  # <-- specified folder id
+            & (folders_v2.c.product_name == product_name)
```

Comment:
YOu see, the fact that you do not passed the `user_id` was already telling me that you assume here that you do no have the rights to access this information